dask/base.py

```diff
@@ -95,49 +125,126 @@ def compute(self, **kwargs):
+    subclass from ``DaskMethodsMixin`` if desired."""
+    __slots__ = ()
+
     @classmethod
-    def _get(cls, dsk, keys, get=None, **kwargs):
-        get = get or _globals['get'] or cls._default_get
-        dsk2 = optimization_function(cls)(ensure_dict(dsk), keys, **kwargs)
-        return get(dsk2, keys, **kwargs)
+    def __subclasshook__(cls, other):
+        if cls is Base:
+            warnings.warn("DeprecationWarning: `dask.base.Base` is deprecated. "
+                          "To check if an object is a dask collection use "
+                          "dask.base.is_dask_collection.")
+        return NotImplemented
+
+    def __dask_graph__(self):
+        # We issue a deprecation warning for the whole class here, as any
+        # non-instance check usage will end up calling `__dask_graph__`.
+        warnings.warn("DeprecationWarning: `dask.base.Base` is deprecated. "
+                      "To create a custom dask object implement the dask "
+                      "collection interface, and optionally subclass from "
+                      "``DaskMethodsMixin`` if desired.")
+        return self.dask
+
+    def _keys(self):
+        warnings.warn("DeprecationWarning: the `_keys` method is deprecated, "
+                      "use `__dask_keys__` instead")
+        return self.__dask_keys__()
+
+    @property
+    def _finalize(self):
+        warnings.warn("DeprecationWarning: the `_finalize` method is "
+                      "deprecated, use `__dask_postcompute__` instead")
+        f, args = self.__dask_postcompute__()
+        return partial(call_finalize, f, args) if args else f
 
     @classmethod
-    def _bind_operator(cls, op):
-        """ bind operator to this class """
-        name = op.__name__
+    def _optimize(cls, *args, **kwargs):
+        warnings.warn("DeprecationWarning: the `_optimize` method is "
+                      "deprecated, use `__dask_optimize__` instead")
+        return cls.__dask_optimize__(*args, **kwargs)
 
-        if name.endswith('_'):
-            # for and_ and or_
-            name = name[:-1]
-        elif name == 'inv':
-            name = 'invert'
+    @classmethod
+    def _get(cls, dsk, keys, **kwargs):
+        warnings.warn("DeprecationWarning: the `_get` method is "
+                      "deprecated, use ``dask.base.compute_as_collection`` "
+                      "instead")
+        return compute_as_collection(cls, dsk, keys, **kwargs)
 
-        meth = '__{0}__'.format(name)
 
-        if name in ('abs', 'invert', 'neg', 'pos'):
-            bind_method(cls, meth, cls._get_unary_operator(op))
-        else:
-            bind_method(cls, meth, cls._get_binary_operator(op))
+def compute_as_collection(cls, dsk, keys, get=None, **kwargs):
+    """Compute a graph as if it were of type cls.
 
-            if name in ('eq', 'gt', 'ge', 'lt', 'le', 'ne', 'getitem'):
-                return
+    Allows for applying the same optimizations and default scheduler."""
```

Comment:
This name didn't make sense to me at first.  I thought it meant something more like "compute this but then return a dask collection".  I don't yet have an idea for a better name though.

Reply:
As the name doesn't show up in the PR, but is in the file diff, adding the name to the comments to make it easier to follow. The name is `compute_as_collection`.

Reply:
Perhaps `compute_as_if_collection`?